### ISSUE TITLE
Commander makes the CLI untestable

### ISSUE THREAD

**Issue by nathankleyn** [shantytown/shanty#4]
Commander appears to use global state for commands. This means that once you call `run!` using Commander, any tests that run after that don't re-run the Commander code correctly. This makes it impossible to test all the scenarios we need to test within the CLI: any success causes any future `ARGV` changes to be ignored.

We need to switch to using something that can be tested, as this isn't an acceptable state to release 1.0 in.

**Comment by nathankleyn** [comment]
I have done some research into this, and come up with a few viable options for replacements:

* https://github.com/leejarvis/slop: Slop is simple, lightweight, has no dependencies, and does only what we need it to and nothing more. It does not support subcommands like Commander does, but that's something we can handle easily ourselves (`ARGV.shift`, check whether command exists, call it or fail, done). The quality of the code in Slop is incredible, so it's a dependency I have confidence in.
* https://github.com/mdub/clamp: Clamp provides a more framework approach to CLI parsing, like Commander. It provides support for subcommands, among other things. What it does not do is make it easy to use in our use-case. We'd have to do a lot of metaprogramming to construct the commands the way Clamp needs them constructed.
* https://github.com/manageiq/trollop: Another simple, lightweight CLI parsing framework, but everything feels a bit manual and it doesn't do much itself. It's the bare-metal approach of the three, and for that reason we probably don't want to go with this or we'll end up writing quite a bit of code we don't want to write.

For the reasons above, I'm going to proceed with replacing Commander with Slop, and finishing off our 100% test-coverage.

**Comment by nathankleyn** [comment]
I went on an exploration to try and get everything working with Slop, and it didn't go great: Slop v4 is what we'd want to use, and it causes conflicts with Pry/ByeBug and only works on Ruby 2.0. Slop v3 isn't exactly what I want to use.

In the meantime, I decided to see if I could get Commander working, and I figured out what was making the tests not work. So, thus, we can stick with Commander for now.
